File: Golang_src/easy/missingNumber_268.go
package easy

// Use Bit Manipulation:
// Time Complexity: O(n)
// Space Complexity:O(1)
// Runtime: 12 ms, faster than 99.34%
// Memory Usage: 6 MB, less than 100.00%
func missingNumber(nums []int) int {
	size := len(nums)
	result := 0

	for i := 0; i < size; i++ {
		result ^= i
		result ^= nums[i]
	}

	result ^= size

	return result
}

File: C_src/Easy/268_Missing_Number/MissingNumber_268.c
#include <stdio.h>
#include <stdlib.h>

// Use bit manipulation
// Time Complexity: O(n)
// Space Complexity:O(1)
// Runtime: 16 ms, faster than 95.72%
// Memory Usage: 7.9 MB, less than 71.43%
int missingNumber(int *nums, int numsSize)
{
    int result = numsSize;
    int i;
    for (i = 0; i < numsSize; i++)
    {
        result ^= i;
        result ^= nums[i];
    }
    return result;
}